pages/posts/flight-of-a-byte.tsx

```diff
@@ -0,0 +1,112 @@
+                </p>
+                <p>
+                The message store receives (code) the event for a new head which contains the <code>CID</code> of the message. The message (which may be large) is fetched with IPFS, not with libp2p pubsub (which is better for small messages). If you don&apos;t know what a <code>CID</code> is and how IPFS uses it to fetch content, then you can read all about it. Notice that using a <code>CID</code> to fetch the message (code) like “Hello” is a separate process from synchronizing heads. Maybe IPFS has already fetched the same <code>CID</code>, or maybe it can be fetched from another group member that is not <code>client2</code>. Wesh takes advantage of the efficient IPFS algorithms.
+                </p>
+                <h3>Decrypting and finishing</h3>
+                <p>
+                We&apos;re almost done! When the “Hello” message is fetched, it&apos;s placed in the local copy of the message log in the correct order. (Messages in the device&apos;s storage remain encrypted.) This internally sends the EventReplicated event which is received (code) by the Wesh message monitor. We started at the libp2p “level” which doesn&apos;t care about the meaning of pubsub messages, up through the message log level which only cares about the correct order of messages. Now at the “top” level, Wesh must use <code>client1&apos;s</code> cryptographic keys to decrypt (code) the message.
+                </p>
+                <p>
+                The decrypted message is sent internally (code) to all monitoring processes, including the GroupMessageList service (code). The <code>client1</code> application has called GroupMessageList, so the “Hello” message is sent (code) to the application which prints the message.
+                </p>
+                <h3>Flight of a Byte</h3>
+                <p>
+                In summary, the flight of a message from <code>client2</code> to client1 starts when the <code>client2</code> peer is discovered, subscribes to the libp2p pubsub topic for the group messages and publishes its message log head. <code>client1</code> doesn&apos;t have this head yet, so it uses the CID to fetch the message using IPFS. When the message arrives, it is added to <code>client1&apos;s</code> copy of the message log in the correct order. Wesh sends this new message to processes using the GroupMessageList service, such as the sample application which receives it and prints “Hello”. Easy, right? Each of these steps is needed to make sure communication can still happen even if a peer disconnects or reconnects at a different network address.
+                </p>
```

Comment:
I'm not sure why client2 is in code font but client1 isn't. Would it be possible to be consistent?

Reply:
done.